crates/ide-completion/src/tests/expression.rs

```diff
@@ -651,3 +651,22 @@ fn main() {
         "]],
     );
 }
+
+#[test]
+fn complete_record_expr_path() {
+    check(
+        r#"
+struct Zulu;
+impl Zulu {
+    fn test() -> Self { }
+}
+fn boi(val: Zulu) { }
+fn main() {
+    boi(Zulu:: $0 {});
+}
+"#,
+        expect![[r#"
+            fn test() fn() -> Zulu
```

Comment:
Can you add a FIXME here, we shouldn't be completing functions in this position, only structs, enum variants and unions

Reply:
Actually this was the reason for opening this PR in the first place. I initially had code like:

```rust
fn main() {
    boi(Zulu { ... });
}
```

But I wanted to search for a method on `Zulu` instead, so instead of actually deleting the `{ ... }`, I started typing typing `Zulu::<|>`. Before this PR this wouldn't have worked.

Reply:
Mmh, ye I think this might be us doing too much if we get rid of those completions, you are right. We should probably look into prioritizing records though, not relevant here now though 👍